pallets/pool-system/src/lib.rs

```diff
@@ -1114,6 +1109,7 @@ pub mod pallet {
 			Ok(())
 		}
 
+		// This method have to be always called from a transaction
```

Comment:
This method makes calls to other pallets that modify their own storage, and it could fail in the middle of the process. If it happens, some data could remain changed on those pallets after the failure.

Those who use this method, should use transaction, i.e, the `execute_epoch` call.

Reply:
I think since Substrate 0.9.26 or similar, all extrinsics are transactional by default.

Reply:
Didn't know that, but if it is the case then we are good. Otherwise, I would add a `#[transactional]` to the method.. 

We should have a `assert_noop` test in this case in the integration tests.